### PR TITLE
m_field_info 37 additional functions OK (40/188 total)

### PR DESCRIPTION
**Pull request by kuchenmitsahne**
### TL;DR

This is a partial decomp of `m_field_info` (37 additional functions OK; 40/188 total matching). Pretty much everything is copy-paste from ac-decomp with minor fix-ups. Verifying the related enums and defines in `m_field_make.h` and `m_random_field.h` was a bit more involved, and I've discussed this in detail below, but ultimately they are very similar to ac-decomp as well.

#### Style

Since most functions match without any changes, the changes are mostly stylistic:

* Changed most local variables to be more verbose (e.g. `blockX` instead of `bx`) and use `camelCase` instead of `snake_case`
* Changed all accesses of the form `bx[0]` to simply `*bx` since the functions seem to only ever expect a pointer to a single entity rather than an array

#### Matching quirks

Explicit `== TRUE` required for `mFI_GetBlockXMax` and `mFI_GetBlockZMax`

#### Differences with ac-decomp

* In `mFI_GetFieldId` the usual `if (mFI_CheckFieldData() == TRUE)` guard clause is omitted, so a null pointer access could potentially occur.

* `BLOCK_Z_NUM` is actually 8 rather than 10 since there's no island in AF.

* `mFM_BLOCK_TYPE_RIVER_SOUTH_CLIFF_HORIZONTAL` in both AC and AF has the `RIVER1` type and (according to ACSE) flows east as one would expect, not south. Flowing south over a horizontal cliff doesn't make sense, since this would instead be a waterfall type. Making this `EAST_CLIFF` normalizes the two river cliff categories (three each for east and west: the two bottom corner types and the horizontal type).

* Besides the consistent differences described below, there was one thing that struck me as odd: `mFM_BLOCK_TYPE_WATERFALL_STRAIGHT_CLIFF_BOTTOM_RIGHT_CORNER` includes a blockkind of `RIVER1` in AF but `RIVER0` in AC. I suspect this is a bug in AF which was corrected in AC. `RIVER0` is used for blocks where the river flows south through the entire block, while `RIVER1` is used for blocks where the river flows east through the entire block. I verified that (according to ACSE) all acres with the `mFM_BLOCK_TYPE_WATERFALL_STRAIGHT_*` block types river flows south, so only `RIVER0` should be used. I have not actually imported the `m_random_field` data I referenced in this PR as it would be mostly unrelated to the core `m_field_info` changes, but when I do I'll mention this in a brief comment.

##### Block kinds

In AF, there is a set of 7 block kinds for the various different cliff blocks analogous to the 7 river block kinds. AC shifts the river block kinds down and does away with the cliff block kinds, instead using those bits for new features such as the museum, the Abel sisters, and the island.

##### Block types

There are fewer block types compared to AC on account of those missing features, although the new block types in AC are placed after the AF block types so the block types match one-to-one up to that point. I mostly left the names the same as in `ac-decomp` with the following exceptions:

###### Different compositions

There are two block types (`SHOP` and `POST_OFFICE`) which have the `RAILROAD` blockkind in AC but not AF; I removed `TRACKS` from those names. AC doesn't use the cliff block kinds as previously mentioned, but `CLIFF` is still used in all relevant block type names so no change was necessary.

###### Endianness

I made the block type names consistently big-endian. For example, `CLIFF_TOP_CORNER_RIGHT` became `CLIFF_CORNER_RIGHT_TOP` since:

* There are three varieties of cliff (corner, vertical, horizontal)
* Of the three, two are differentiated by left/right (corner, vertical)
* The corner cliffs each have two corresponding blocks (top and bottom)

###### Straight -> South

Pretty self explanatory--the waterfall block types in AC use `STRAIGHT` instead of `SOUTH`, which is confusing since it still uses `EAST` and `WEST` like the river blockt types (and further confusing since the actual waterfalls are often diagonal). I changed them to use `SOUTH`.

#### Challenges

Verifying most of the block types was fairly trivial--I would simply create a new town, use a script to extract the block types, and if there were types I still needed to verify present I would go ahead and finish the opening sequence and inspect them in-game. However, the waterfalls proved difficult, as it seems only a few of them appear regularly and the town generation logic was too complicated for me to figure out how to force specific ones to appear without ending up with an invalid map that would cause the game to hang.

To make a long story short, I pored over the code in `ac_decomp` and stepped through the debugger for a while before determining that `data_combi_table` held the key. `data_combi_table` is an array of structs which hold the fgID, bgID, and block type of every single acre type. When a save is loaded, `mFM_SetBlockKind` is called to fetch the block type of each acre, fetch the block kinds of that block type, and store each in arrays pointed to by `g_block_type_p` and `g_block_kind_p`, respectively. It uses the acre ID to index into `data_combi_table`, but crucially, that acre ID is stored in a `u16` bitfield with 14 bits, so an acre ID of `0x006C` in memory actually corresponds to an index of `0x001B` (i.e. `(0x006C >> 2)`). Once I had that figured out and AF's `data_combi_table` extracted (different fg and bg IDs, but seemingly identical blockt type information as AC other than being shorter), I used Cuyler's ACSE to get the acre IDs and bird's eye images of all 15 waterfall acre variants and checked their block types in `data_combi_table` to make sure none of them deviated from the block type name.

### CODE REVIEW

**Comment by kuchenmitsahne** [comment]
Of _course_ I didn't notice that I put `FieldMakeBlockType` in the wrong header file until after I open the PR! While fixing that I noticed and removed some extraneous defines that don't appear to be used anywhere else and are covered by the ones added by this PR, so there's a silver lining.

**Review by emilybrooks — APPROVED** [review]
